go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -162,5 +183,134 @@ func initLocalnetGatewayInternal(nodeName string, clusterIPSubnet []string,
 			err)
 	}
 
+	if nodePortEnable {
+		return localnetNodePortWatcher(ipt, wf)
+	}
+
 	return nil
 }
+
+// AddService adds service and creates corresponding resources in OVN
+func localnetAddService(svc *kapi.Service) error {
+	if svc.Spec.Type != kapi.ServiceTypeNodePort {
+		return nil
+	}
+	ipt, err := iptables.NewWithProtocol(iptables.ProtocolIPv4)
+	if err != nil {
+		return fmt.Errorf("failed to initialize iptables: %v", err)
+	}
+	rules := make([]iptRule, 0)
+	for _, svcPort := range svc.Spec.Ports {
+		protocol := svcPort.Protocol
+		if protocol != kapi.ProtocolUDP && protocol != kapi.ProtocolTCP {
```

Comment:
Doesn't this prevent SCTP which is available in Kube as of 1.13 or something like that?

Reply:
Yes. However, this is a generic issue across all gateway modes we support today. So, we will need to add support for SCTP separately.